e2e/e2e_test.go

```diff
@@ -3,89 +3,61 @@ package whereabouts_e2e
 	"context"
 	"encoding/json"
+	"fmt"
 	"net"
 	"os"
+	"strings"
 	"testing"
-
 	"time"
 
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
 	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
 	netclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
 	core "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-// Global Constants
 const (
 	testNetworkName = "wa-nad"
 	testNamespace   = "default"
 	testImage       = "quay.io/dougbtv/alpine:latest"
 	ipv4TestRange   = "10.10.0.0/16"
 	singlePodName   = "whereabouts-basic-test"
+	createTimeout   = 10 * time.Second
+	deleteTimeout   = 2 * createTimeout
 )
 
-// ClientInfo contains information given from k8s client
-type ClientInfo struct {
-	Client    kubernetes.Interface
-	NetClient netclient.K8sCniCncfIoV1Interface
-}
-
 func TestWhereaboutsE2E(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "whereabouts-e2e")
 }
 
 var _ = Describe("Whereabouts functionality", func() {
 	Context("Test setup", func() {
-		// Declare variables
 		var (
-			err          error
-			kubeconfig   string
-			label        map[string]string
-			annotations  map[string]string
-			clientInfo   ClientInfo
-			clientSet    *kubernetes.Clientset
-			netClient    netclient.K8sCniCncfIoV1Interface
+			clientInfo   *ClientInfo
 			netAttachDef *nettypes.NetworkAttachmentDefinition
-			config       *rest.Config
-			netStatus    []nettypes.NetworkStatus
 			pod          *core.Pod
 		)
 
 		BeforeEach(func() {
-			var found bool
-			kubeconfig, found = os.LookupEnv("KUBECONFIG")
-			Expect(found).To(BeTrue(), "must provide the path to the kubeconfig via the `KUBECONFIG` env variable")
-
-			config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
-			Expect(err).To(BeNil())
-
-			// Create k8sclient and ClientInfo object
-			clientSet, err = kubernetes.NewForConfig(config)
-			Expect(err).To(BeNil())
-			netClient, err = netclient.NewForConfig(config)
-			Expect(err).To(BeNil())
-
-			clientInfo = ClientInfo{
-				Client:    clientSet,
-				NetClient: netClient,
-			}
-			netAttachDef = macvlanNetworkWithWhereaboutsIPAMNetwork()
+			config, err := clusterConfig()
+			Expect(err).NotTo(HaveOccurred())
+
+			clientInfo, err = NewClientInfo(config)
+			Expect(err).NotTo(HaveOccurred())
 
-			label = make(map[string]string)
-			annotations = make(map[string]string)
-			annotations["k8s.v1.cni.cncf.io/networks"] = testNetworkName
+			netAttachDef = macvlanNetworkWithWhereaboutsIPAMNetwork()
 
-			// Create a net-attach-def
 			By("creating a NetworkAttachmentDefinition for whereabouts")
 			_, err = clientInfo.addNetAttachDef(netAttachDef)
-			Expect(err).To(BeNil())
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Why change from `To(BeNil())` to `NotTo(HaveOccurred())` ? 
`To(BeNil())` reads more clearly without using negatives. Very pedantic comment, but I used `To(BeNil())` initially on your recommendation, so this is leaving me confused. Ditto on all other occurrences in this PR.

Reply:
erm, yeah, if I said that, I was wrong. My bad then.

You should error check with `Expect(err).To(Succeed())` or `Expect(err).To(HaveOccurred())`:
https://github.com/onsi/gomega/blob/7e687ad94511b02c75ab34d08e02c5b1a68358b3/matchers.go#L53
> //HaveOccurred succeeds if actual is a non-nil error
//The typical Go error checking pattern looks like:
//    err := SomethingThatMightFail()
//    Expect(err).ShouldNot(HaveOccurred())

One thing is a non-nil thing, another a non-nil error. Same thing with `Succeed()` which is a few lines below.

Reply:
Possibly me being confused here as well, no worries. Thanks for the explanation